tools/syz-repro/repro.go

```diff
@@ -76,15 +78,26 @@ func main() {
 	}
 
 	fmt.Printf("opts: %+v crepro: %v\n\n", res.Opts, res.CRepro)
-	fmt.Printf("%s\n", res.Prog.Serialize())
+
+	if progSerialized := res.Prog.Serialize(); len(progSerialized) > 0 {
+	  fmt.Printf("%s\n", progSerialized)
+	  if err := ioutil.WriteFile(*flagOutput, progSerialized, 0644); err == nil {
+	    log.Logf(0, "Failed to write prog to file: %v", err)
+	  }
+	}
+
 	if res.CRepro {
 		src, err := csource.Write(res.Prog, res.Opts)
 		if err != nil {
 			log.Fatalf("failed to generate C repro: %v", err)
 		}
 		if formatted, err := csource.Format(src); err == nil {
-			src = formatted
-		}
-		fmt.Printf("%s\n", src)
+		  if len(formatted) > 0 {
```

Comment:
This check is unnecessary. If programs are transformed to empty C programs, we have bigger problems and it wouldn't compile, so it can't be a reproducer.

Reply:
from my test run, I have gotten a non empty program with empty C program. Should I log (or error) in this case?